hs/src/Reach/Backend/JS.hs

```diff
@@ -709,7 +714,11 @@ jsPIProg cr (PLProg _ (PLOpts {}) dli (EPPs pm) _) = do
   partsp <- mapM (uncurry (jsPart dli)) $ M.toAscList pm
   cnpsp <- mapM (uncurry jsCnp) $ HM.toList cr
   let connsExp = jsConnsExp $ HM.keys cr
-  return $ vsep_with_blank $ preamble : emptyDoc : partsp ++ emptyDoc : cnpsp ++ [emptyDoc, connsExp, emptyDoc]
+  exportProps <- mapM (\ (k, v) -> do
+        vs <- jsExportValue v
+        return $ "    " <> pretty k <> " : " <> vs <> ",\n") dexports
+  let exportsp = "export const getExports = (stdlib) => {\n  return ({\n" <> hcat exportProps <> "  });\n};"
```

Comment:
This is going to have to have its own "contract environment" like individual backends have